src/python/gudhi/hera.py

```diff
@@ -65,11 +66,24 @@ def wasserstein_distance(
             convention that (-1) represents the diagonal. If the distance between two diagrams is +inf (which happens
             if the cardinalities of essential parts differ) and the matching is requested, it will be set to ``None``
             (any matching is optimal).
+            
+            .. warning::
+            
+                For matching request, please consider using :func:`~gudhi.wasserstein.wasserstein_distance` (POT
+                version) instead. This version with ``matching=True`` is known to have bugs.
 
     Returns:
         float|Tuple[float,numpy.array|None]: Approximate Wasserstein distance W_q(X,Y), and optionally the 
             corresponding matching
+    
     """
+    if matching:
+        warnings.warn(
+            """
+            There is known bug (https://github.com/GUDHI/gudhi-devel/issues/881) when `matching` is set to `True` with
+            the Hera backend. We recommend using `gudhi.wasserstein.wasserstein_distance` instead.
```

Comment:
I'd be tempted to add "for now" or "for the moment" at the end, to clarify that this is temporary, but ok with or without that change.

Reply:
Done on [414be1d](https://github.com/GUDHI/gudhi-devel/pull/1245/commits/414be1dac0f0ff87ecb68a089a8166e1be5eab51) + [e0cc1f5](https://github.com/GUDHI/gudhi-devel/pull/1245/commits/e0cc1f5ffbdf76080c1981ff55763b85fb4689b6)